test/tests/functional/pbs_dup_acc_log_for_resv.py

```diff
@@ -47,16 +47,298 @@ class TestDupAccLogForResv(TestFunctional):
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using America/Los_Angeles')
+            tzone = 'America/Los_Angeles'
+        a = {'Resource_List.select': '2:ncpus=1',
+             'Resource_List.place': 'scatter',
+             ATTR_resv_rrule: 'FREQ=MINUTELY;COUNT=3',
+             ATTR_resv_timezone: tzone,
+             ATTR_resv_standing: '1',
+             'reserve_start': time.time() + 5,
+             'reserve_end': time.time() + 60, }
+        r = Reservation(TEST_USER, a)
+        rid = self.server.submit(r)
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, max_attempts=30, offset=6)
+        if _m == PTL_API:
+            self.server.set_op_mode(PTL_API)
+
+        # Verify accounting log before server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+        # Restart server
+        self.server.restart()
+
+        # Verify accounting log after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, max_attempts=30, offset=60)
+
+        # Verify accounting log for standing reservation starting second time
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 2)
+
+        # Restart server
+        self.server.restart()
+
+        # Verify accounting log after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 2)
+
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, max_attempts=30, offset=60)
+
+        # Verify accounting log for standing reservation starting third time
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 3)
+
+        # Restart server
+        self.server.restart()
+
+        # Verify accounting log after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 3)
+
+    def test_accounting_logs_after_qterm(self):
+        """
+        Test for duplicate records in accounting log for reservationsWhen
+        when server stopped using qterm and start again.
```

Comment:
a space between reservations and When is needed. There are two "when"s in the comment.
A little rephrasing - when the server is stopped using qterm and started again.